modules/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolEnter.java

```diff
@@ -359,4 +377,101 @@ public void defineNode(BLangNode node, SymbolEnv env) {
         node.accept(this);
         this.env = prevEnv;
     }
+
+    private void defineConnectorInitFunction(BLangConnector connector) {
```

Comment:
Looks good. 

But seems like we have common logic repeated in all three methods. Can we refactor them?

Reply:
Fixed with 6a6642b8094d731de9e332c0f6b3a52a7b54bfcc